### PR TITLE
Change log(x+1) to log1p(x) in LogTransformer

### PR DESCRIPTION
**Pull request by ChaitanMohr**
changed instances of log(x + 1) to log1p(x) which is better conditioned for small x

### CODE REVIEW

**Review by davidhopkinson26 — CHANGES_REQUESTED** [review]
simple change and happy with it.  

Could you please add a section to the changelog and increment version.py to 1.2.2.  If you look at the raw files for the Changelog.md you should be able to see the correct format to link in the PR and format the section headers etc.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -16,6 +16,13 @@ Subsections for each version can be one of the following;
 
 Each individual change should have a link to the pull request after the description of the change.
 
+1.2.2 (2024-02-15)
+------------------
+Changed
+^^^^^^^
+- Changed LogTransformer to use log1p(x) instead of log(x+1) `#178 <https://github.com/lvgig/tubular/pull/178>`_
+- Changed unit tests using log(x+1) to log1p(x) `#178 <https://github.com/lvgig/tubular/pull/178>`_
+
```
Comment (davidhopkinson26) [inline CHANGELOG.rst:25]: Also looks like the last release went out without updating the changelog.  Could you update for 1.2.1 too please?

Comment (ChaitanMohr) [reply CHANGELOG.rst:25]: looking at the merge conflicts there appears to have been the 1.2.1 changelog added at some point

**Review by davidhopkinson26 — APPROVED** [review]
looks good! happy with changes

**Pull request merged**